src/main/java/de/craften/ui/swingmaterial/MaterialButton.java

```diff
@@ -179,7 +222,7 @@ protected void paintComponent(Graphics g) {
         g2.drawString(getText().toUpperCase(), x, y);
 
         if (isEnabled()) {
-            g2.setClip(new RoundRectangle2D.Float(0, 0, getWidth() - offset_lr, getHeight() - offset_td, 3, 3));
+            g2.setClip(new RoundRectangle2D.Float(0, 0, getWidth() - offset_lr, getHeight() - offset_td, Math.max(borderRadius*2 - 4, 0), Math.max(borderRadius*2 - 4, 0)));
```

Comment:
That's for the ripple to not be left behind the button borders. As only the buttons use them *this* way, I see no reason to use a constant. If this were to change though, a constant pool should be considered for other classes to reuse this (and maybe some other constants could be pulled out from their current classes then).